Lib/test/test_ast/test_ast.py

```diff
@@ -298,7 +298,7 @@ def test_arguments(self):
         x = ast.arguments()
         self.assertEqual(x._fields, ('posonlyargs', 'args', 'vararg', 'kwonlyargs',
                                      'kw_defaults', 'kwarg', 'defaults'))
-        self.assertEqual(x.__annotations__, {
```

Comment:
What is the behavior of accessing `__annotations__` on an instance of a class, instead of the class itself, with this PR?

Reply:
It shouldn't work (unless the user manually added `__annotations__` to the class namespace). On current main it instead behaves erratically; it works only if the `__annotations__` have been cached in the class dict.